python/vowpalwabbit/dftovw.py

```diff
@@ -702,10 +702,10 @@ def __init__(
     @classmethod
     def from_colnames(
         cls,
-        y: Union[Hashable, List[Hashable]],
-        x: Union[Hashable, List[Hashable]],
         df: pd.DataFrame,
-        label_type: str = "simple_label",
+        x: Union[Hashable, List[Hashable]],
+        y: Optional[Union[Hashable, List[Hashable]]] = None,
+        label_type: Optional[str] = "simple_label",
```

Comment:
This change breaks existing code which uses this. While less ergonomic if you just make `y` optional and dont reorder the arguments then existing usage should be unaffected.

Reply:
I agree

Reply:
>This change breaks existing code which uses this.

This is technically true. However it would only break the code that uses positional arguments instead of keyword arguments. I've looked at all GitHub code that uses `from_colnames` and it looks like all calls use keyword arguments.
https://github.com/search?q=from_colnames&type=code

>if you just make y optional and dont reorder the arguments

Unfortunately, Python does not allow optional parameters before required parameters.
So, I cannot make `y` optional without making `x` and `df` optional. // In the future, we should try to put "more required" parameters first.

The only other solution I can think of is pretty ugly: leave `y` required, but make it accept `None`.
What do you think?

Reply:
> This is technically true. However it would only break the code that uses positional arguments instead of keyword arguments. I've looked at all GitHub code that uses from_colnames and it looks like all calls use keyword arguments.

I know this seems minor and probably is low impact but we're making a conscious effort to have a more stable API so I think we should adhere to that here. 

> The only other solution I can think of is pretty ugly: leave y required, but make it accept None.
What do you think?

Sorry, when I said make it optional what I really meant was what you described. I agree it's ugly, but I can't think of another way to enable this in a non-breaking way.